win2xcur/multiscale.py

```diff
@@ -0,0 +1,32 @@
+from win2xcur.cursor import CursorFrame, CursorImage
+
+MULTSCALE = [16, 24, 32, 48, 64, 96, 128, 192, 256]
+
+def generates_frames(cursor, min_size: int) -> None:
+    """Generates multiple sizes for each cursor.
+
+    Args:
+        cursor (Cursor): The cursor to generate sizes for.
+        min_size (int): The minimum size to generate.
+
+    Returns:
+        List[Cursor]: The generated cursors.
+    """
+    frames = cursor.frames
+    new_frames = []
+    image_size = frames[0].images[0].nominal
+    for size in MULTSCALE:
+        if size > image_size:
+            continue
+        if size < min_size:
+            break
+        for frame in frames:
+            new_images = []
+            for cur in frame:
+                new_cur = CursorImage(cur.image.clone(), cur.hotspot, cur.nominal)
+                new_cur.scale(size, size)
+                new_images.append(new_cur)
+            new_frame = CursorFrame(new_images, frame.delay)
```

Comment:
Why are you looping over all the sizes and repeat each frame for each size? I see why you run into the nominal size problem now. If the nominal size is different in different frames, then Xcursor treats them as separate sizes and not animations. This is a sign of a massive bug in your implementation—the number of frames should be constant despite all your image manipulation.